tests/unit/modules/test_chroot.py

```diff
@@ -182,3 +182,35 @@ def test_call_success(self, mkdtemp, exist):
             salt_mock['archive.tar'].assert_called_once()
             salt_mock['cmd.run_chroot'].assert_called_once()
             utils_mock['files.rm_rf'].assert_called_once()
+
+    @patch('salt.modules.chroot.exist')
+    @patch('tempfile.mkdtemp')
+    def test_call_success_parameters(self, mkdtemp, exist):
+        '''
+        Test execution of Salt functions in chroot with parameters.
+        '''
+        # Success test
+        exist.return_value = True
+        mkdtemp.return_value = '/chroot/tmp01'
+        utils_mock = {
+            'thin.gen_thin': MagicMock(return_value='/salt-thin.tgz'),
+            'files.rm_rf': MagicMock(),
+            'json.find_json': MagicMock(return_value={'return': 'result'})
+        }
+        salt_mock = {
+            'archive.tar': MagicMock(return_value=''),
+            'config.option': MagicMock(),
+            'cmd.run_chroot': MagicMock(return_value={
+                'retcode': 0,
+                'stdout': '',
+            }),
+        }
+        with patch.dict(chroot.__utils__, utils_mock), \
+                patch.dict(chroot.__salt__, salt_mock):
+            self.assertEqual(chroot.call('/chroot', 'ssh.set_auth_key',
+                                         user='user', key='key'), 'result')
+            utils_mock['thin.gen_thin'].assert_called_once()
+            salt_mock['config.option'].assert_called()
+            salt_mock['archive.tar'].assert_called_once()
+            salt_mock['cmd.run_chroot'].assert_called_once()
+            utils_mock['files.rm_rf'].assert_called_once()
```

Comment:
I'm not sure that I understand - what is it that this is testing here?

Reply:
`test_call_success_parameters` is testing that the call succeed when args and kwargs are provided, that is where the bug was living.

Those asserts validate that the mocks are called, so we are sure that the full process takes place (create the salt-thin, the call to cmd.run_chroot, the clean at the end, etc)

Reply:
I suppose that's reasonable. I'd be happier if we had a functional test (i.e. a test that calls `chroot.call('/chroot', 'ssh.set_auth_key', user='user', key='key')`, and it really creates the chroot & does the filesystem stuff), but it's *probably* fine to test like this 🤷‍♂